src/culture/rebels.cpp

```diff
@@ -771,9 +771,90 @@ void daily_update_rebel_organization(sys::state& state) {
+			rebel_hunters.clear();
+			for(auto ar : faction_owner.get_army_control()) {
+				auto loc = ar.get_army().get_location_from_army_location();
+				if(ar.get_army().get_is_rebel_hunter()
+					&& !ar.get_army().get_battle_from_army_battle_participation()
+					&& !ar.get_army().get_navy_from_army_transport()
+					&& (ar.get_army().get_arrival_time() || loc.get_nation_from_province_control() == faction_owner)
+					) {
+
+					rebel_hunters.push_back(ar.get_army().id);
+				}
+			}
+
+			static std::vector<dcon::province_id> rebel_provs;
+			rebel_provs.clear();
+			for(auto prov : faction_owner.get_province_ownership()) {
+				if(prov.get_province().get_rebel_faction_from_province_rebel_control()
+					|| military::rebel_army_in_province(state, prov.get_province()))
+					rebel_provs.push_back(prov.get_province().id);
+			}
+
+			while(rebel_provs.size() > 0 && rebel_hunters.size() > 0) {
+				auto rh = rebel_hunters[0];
+				auto loc = state.world.army_get_location_from_army_location(rh);
+				std::sort(rebel_provs.begin(), rebel_provs.end(), [&](dcon::province_id a, dcon::province_id b) {
+					auto da = province::sorting_distance(state, a, loc);
+					auto db = province::sorting_distance(state, b, loc);
+					if(da != db)
+						return da < db;
+					else
+						return a.index() < b.index();
+				});
+
+				auto closest_prov = rebel_provs[0];
+				std::sort(rebel_hunters.begin(), rebel_hunters.end(), [&](dcon::army_id a, dcon::army_id b) {
+					auto pa = state.world.army_get_location_from_army_location(a);
+					auto pb = state.world.army_get_location_from_army_location(b);
+					auto da = province::sorting_distance(state, pa, closest_prov);
+					auto db = province::sorting_distance(state, pb, closest_prov);
+					if(da != db)
+						return da < db;
+					else
+						return a.index() < b.index();
+				});
+
+				for(uint32_t i = 0; i < rebel_hunters.size(); ++i) {
+					auto a = rebel_hunters[i];
+					if(state.world.army_get_location_from_army_location(a) == closest_prov) {
+						auto existing_path = state.world.army_get_path(a);
+						existing_path.resize(0);
+						state.world.army_set_arrival_time(a, sys::date{});
+
+						rebel_hunters[i] = rebel_hunters.back();
+						rebel_hunters.pop_back();
+						break;
+					} else if(auto path = province::make_land_path(state, state.world.army_get_location_from_army_location(a), closest_prov, faction_owner, a); path.size() > 0) {
+						auto existing_path = state.world.army_get_path(a);
+
+						auto new_size = uint32_t(path.size());
+						existing_path.resize(new_size);
+
+						for(uint32_t j = new_size; j-- > 0; ) {
+							existing_path.at(j) = path[j];
+						}
+
+						state.world.army_set_arrival_time(a, military::arrival_time_to(state, a, path.back()));
+						state.world.army_set_dig_in(a, 0);
+
+						rebel_hunters[i] = rebel_hunters.back();
+						rebel_hunters.pop_back();
+						break;
+					}
+				}
+
+				rebel_provs[0] = rebel_provs.back();
+				rebel_provs.pop_back();
```

Comment:
I see some potential issues with this logic. First, while it won't break if pathfinding fails, it may leave some provinces unable to be targeted by rebel hunters. (For example, if the rebel province is nearest to a rebel hunter who is overseas from it, that rebel hunter will not be able to path to it, but the province will still be removed from the list. This could result in provinces that are simply never take back). You should at least consider using the connected region values to filter this check with. Secondly, it doesn't take the size of the rebel stacks into account, and so may result in a suicidal use of rebel hunters, especially if they have become depleted by prior combats. Thirdly, it doesn't prioritize rebel armies over rebel occupations, meaning that you could end up with a rebel hunter forever trailing behind rebel armies, recapturing one province while the rebels capture a new one.